layers/+tools/translate/packages.el

```diff
@@ -0,0 +1,79 @@
+;;; packages.el --- Translate Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Ray Wang <rayw.public@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+(defconst translate-packages
+  '(
+    (translate-mode :location (recipe :fetcher github :repo "rayw000/translate-mode"))
+    go-translate
+    popup
+    posframe
+    ))
+
+(defun translate/init-translate-mode ()
+  "Initialize required packages."
+  (use-package translate-mode
+    :demand t
+    :hook (translate-mode . set-translate-mode-paragraph-functions)))
+
+(defun translate/init-go-translate ()
+  (use-package go-translate
+    :demand t
+    :config
+    (progn
+      (defclass spacemacs//original-paragraph-texter (gts-texter) ())
```

Comment:
those new classes and methods should be moved to funcs.el

Reply:
I've tried before, but it is weird that my `funcs.el` is always loaded before `go-translate` and `translate-mode` packages are ready, differing from described [here](https://develop.spacemacs.org/doc/LAYERS.html#funcsel). Checking `(featurep 'go-translate)` at the beginning of the funcs.el always gets `nil`. That's why I put the class/method definitions into `:config` in `(use-package go-translate)`. Same thing happens in `cmake` layer too. Do you have any idea?

Reply:
I don't think load order is the issue here.

The issue is that when you define the new classes, they inherit from classes defined in the respective packages. So the new class declarations must be evaluated after the loading of the packages.

Okay just keep them here.

Reply:
Yes, I knew your point, and that's what I did at first.
Perhaps there is a bug to be investigated that the actual load order here is not following its documentation. It may not affect function definitions, like the way many other layers use `funcs.el`, but when inheriting from classes, it fails.